src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,147 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAPIException;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+/**
+ * A class that is to be used for boolean expressions.
+ * <p>
+ * A boolean expression should be a condition for getting it's value.
+ */
+public abstract class SetEffect<T> extends Effect {
+
+	private Expression<Boolean> value;
+	private Expression<T> expression;
+	private boolean make, negated;
+
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%" and "set %types%'[s] property to %boolean%"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void register(Class<? extends SetEffect<?>> effect, String property, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%");
+	}
+	
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%", "set %types%'[s] property to %boolean%"
+	 * and "make %types% makeProperty" with "force %types% to makeProperty"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param makeProperty The property of the syntax used for "make %types% makeProperty".
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void registerMake(Class<? extends SetEffect<?>> effect, String property, String makeProperty, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + makeProperty,
+				"force %" + type + "% to " + makeProperty,
+				"make %" + type + "% not " + makeProperty,
+				"force %" + type + "% to not " + makeProperty);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public final boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		make = matchedPattern >= 2;
+		if (exprs.length == 1) {
+			if (!make)
+				throw new SkriptAPIException("There was not two expressions in the SetEffect class '" + getClass().getName() + "' exprs: " + Arrays.toString(exprs));
+			expression = (Expression<T>) exprs[0];
+			negated = matchedPattern >= 4;
+			return true;
+		}
+		expression = (Expression<T>) exprs[matchedPattern];
+		value = (Expression<Boolean>) exprs[matchedPattern ^ 1];
+		return true;
+	}
+
+	/**
+	 * Returns the value of the boolean expression used.
+	 * 
+	 * @param event The event that is calling this syntax.
+	 * @return boolean from the boolean expression used in this set effect.
+	 */
+	@Nullable
+	protected final Boolean getBoolean(Event event) {
```

Comment:
does this really have to be a method? it could probably be combined into the execute call

Reply:
It was to add more functuality to the developer, but I see with it being a bi consumer, it wouldn't matter?